lib/ansible/galaxy/role.py

```diff
@@ -168,12 +179,24 @@ def fetch(self, role_data):
             display.display("- downloading role from %s" % archive_url)
 
             try:
-                url_file = open_url(archive_url)
                 temp_file = tempfile.NamedTemporaryFile(delete=False)
-                data = url_file.read()
-                while data:
-                    temp_file.write(data)
+                if archive_url.startswith('s3:'):
+                    bucket, prefix = parse_s3_url(archive_url)
+                    import boto3
```

Comment:
This needs to be handled better. This import should be at the top of the file and wrapped in a try/except so that boto isn't a hard requirement on ansible-galaxy. Also, this should support both boto and boto3, as both are still in common usage.